### PR TITLE
chore: Revert any failed Vercel API calls

### PR DESCRIPTION
**Pull request by matekdev**

### CODE REVIEW

File: apps/dashboard/src/routes/+page.svelte [hunk 1]
```diff
@@ -89,11 +89,13 @@
 			};
 			const { error: urlError } = await data.supabase.from('urls').insert(newURL);
 			if (urlError) {
-				var msg =
-					urlError.code == '23505'
-						? `URL Prefix with the domain "${subdomain + domain}" already exists!`
-						: toastError.message;
-				toastStore.trigger({ ...toastError, message: msg });
```
Comment (matekdev) [inline apps/dashboard/src/routes/+page.svelte:96]: This error message needs to be moved to the call before. We can do a pass and double check any error UX before release.

File: apps/dashboard/src/routes/+page.svelte [hunk 2]
```diff
@@ -89,11 +89,13 @@
...
+				// Attempt to revert the previous add domain call.
```
Comment (MisterJimson) [inline apps/dashboard/src/routes/+page.svelte:92]: I was thinking we do the revert in the add-domain api-call itself

File: apps/dashboard/src/routes/+page.svelte [hunk 3]
```diff
@@ -126,6 +128,12 @@
 
 			const { error: urlError } = await data.supabase.from('urls').delete().eq('url', selectedURL);
 			if (urlError) {
+				// Attempt to revert the previous delete domain call.
```
Comment (MisterJimson) [inline apps/dashboard/src/routes/+page.svelte:131]: Same here

File: apps/dashboard/src/routes/+page.svelte (hunk 2 above)
Comment (matekdev) [reply apps/dashboard/src/routes/+page.svelte:92]: unless I'm missing something, but how would we know to revert?

If the vercel API call goes through, and the database call doesn't afterwards, then we revert the vercel API call. We need to know the status of the database call.

Are you suggesting that we move the database call inside of the vercel API call?

**Comment by matekdev** [comment]
Let's chat about this at a later point since there are a bunch of merge conflicts.

**Pull request closed without merging**